galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt

```diff
@@ -15,6 +15,2538 @@
+            if (query.isNull(1)) {
+              continue
+            }
+            var i = 0
+            while (i < lines.size) {
+              lines[i][lineCount] = query.getObject(i + if (SKIP_FIRST_COLUMN) 2 else 1)
+              i += 1
+            }
+            lineCount += 1
+          }
+          query.close()
+          getForm().commitProtected()
+          break
+        } catch (e: SQLException) {
+          getForm().abortProtected(e)
+        } catch (error: Error) {
+          getForm().abortProtected(error)
+        } catch (rte: RuntimeException) {
+          getForm().abortProtected(rte)
+        }
+      }
+    } catch (e: Throwable) {
+      throw VRuntimeException(e)
+    }
+    return if (lineCount == 0 && (newForm == null || !isNull(block!!.getActiveRecord()))) {
+      throw VFieldException(this, MessageCode.getMessage("VIS-00001"))
+    } else {
+      val selected: Int
+
+      selected = if (lineCount == 0 && newForm != null && isNull(block!!.getActiveRecord())) {
+        newForm.add(getForm())
+      } else {
+        if (lineCount == MAX_LINE_COUNT - 1) {
+          getForm().notice(MessageCode.getMessage("VIS-00028"))
+        }
+        if (lineCount == 1 && !SHOW_SINGLE_ENTRY) {
+          0
+        } else {
+          val ld = VListDialog(columns, lines, lineCount, newForm)
+          ld.selectFromDialog(getForm(), null, this)
+        }
+      }
+      if (selected == -1) {
+        throw VExecFailedException() // no message needed
+      } else if (selected >= lineCount) {
+        // new, retrieve it
+        var result: Any? = null
+
+        try {
+          while (true) {
+            try {
+              getForm().startProtected(null)
+              val SELECT_IS_IN_LIST = " SELECT   $1                   " +
+                      " FROM     $2                   " +
+                      " WHERE    $3 = " + selected
+              val query = Query(getForm().dBContext.defaultConnection)
+
+              query.addString(list!!.getColumn(0).column)
+              query.addString(evalListTable())
+              query.addString("ID")
+              query.open(SELECT_IS_IN_LIST)
+              query.next()
+              result = query.getObject(1)
+              query.close()
+              getForm().commitProtected()
+              break
+            } catch (e: SQLException) {
+              getForm().abortProtected(e)
+            } catch (error: Error) {
+              getForm().abortProtected(error)
+            } catch (rte: RuntimeException) {
+              getForm().abortProtected(rte)
+            }
+          }
+        } catch (e: Throwable) {
+          throw VRuntimeException(e)
+        }
+        result
+      } else {
+        lines[0][selected]
+      }
+    }
+  }
+
+  /**
+   * Checks that field value exists in list
+   */
+  protected fun selectFromList(gotoNextField: Boolean) {
+    val qrybuf = StringBuffer()
+
```

Comment:
you can use : 
val qrybuf = buildString{...} and remove qrybuf occurencies inside the scope